programs/svm-spoke/src/instructions/refund_claims.rs

```diff
@@ -102,7 +103,89 @@ pub fn claim_relayer_refund(ctx: Context<ClaimRelayerRefund>) -> Result<()> {
     emit_cpi!(ClaimedRelayerRefund {
         l2_token_address: ctx.accounts.mint.key(),
         claim_amount,
-        refund_address: ctx.accounts.token_account.key(),
+        refund_address: ctx.accounts.signer.key(),
+    });
+
+    // There is no need to reset the claim amount as the account will be closed at the end of instruction.
+
+    Ok(())
+}
+
+#[event_cpi]
+#[derive(Accounts)]
+#[instruction(refund_address: Pubkey)]
+pub struct ClaimRelayerRefundFor<'info> {
+    pub signer: Signer<'info>,
+
+    /// CHECK: We don't need any additional checks as long as this is the same account that initialized the claim account.
+    #[account(mut, address = claim_account.initializer @ SvmError::InvalidClaimInitializer)]
+    pub initializer: UncheckedAccount<'info>,
+
+    #[account(seeds = [b"state", state.seed.to_le_bytes().as_ref()], bump)]
+    pub state: Account<'info, State>,
+
+    #[account(
+        mut,
+        associated_token::mint = mint,
+        associated_token::authority = state,
+        associated_token::token_program = token_program
+    )]
+    pub vault: InterfaceAccount<'info, TokenAccount>,
+
+    // Mint address has been checked when executing the relayer refund leaf and it is part of claim account derivation.
+    #[account(mint::token_program = token_program)]
+    pub mint: InterfaceAccount<'info, Mint>,
+
+    #[account(
+        mut,
+        associated_token::mint = mint,
+        associated_token::authority = refund_address,
```

Comment:
Anchor checks both that ATA is derived from refund_address and that it is owned by refund_address. This differs slightly from the implementation in `execute_relayer_refund_leaf` where we only check ATA address (as owner might have transferred authority), but should not be a big issue as the relayer can recover by calling `claim_relayer_refund` to any custom token account.